### PR TITLE
Replace execution context controls with record mode controls

### LINKED ISSUE #8673: Replace Design/Live/Run control with Record/Record Once Control
- Change the icons on the Design/Live/Run control to the new Record/Record Once icons.
- Functionality should remain unchanged
```[tasklist]
### Tasks
- [ ] https://github.com/enso-org/enso/pull/9133
```

### PR DESCRIPTION
**Pull request by kazcw**
### Pull Request Description

Closes #8673. Stacked on #9132.

Before:
image
image

After:
image
image

### Important Notes

- The model has been refactored from execution-modes (live/design) to record-mode (on/off) up to the point of the project store, where it's translated (since the backend still uses the live/design concepts).

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/gui2/src/assets/icons.svg [hunk 1]
```diff
@@ -650,6 +650,16 @@
...
 
+  <symbol id="record" viewBox="0 0 16 16" width="16" height="16" fill="none">
+    <circle cx="8" cy="8" r="7" fill="currentColor" />
+  </symbol>
+
+  <symbol id="record_once" viewBox="0 0 24 16" width="24" height="16" fill="none">
+    <rect x="0" y="1" width="3" height="14" rx="1" fill="currentColor" />
+    <circle cx="12" cy="8" r="7" fill="currentColor" />
+    <rect x="21" y="1" width="3" height="14" rx="1" fill="currentColor" />
+  </symbol>
```
Comment (farmaazon) [inline app/gui2/src/assets/icons.svg]: Were they added to figma project? If not, then they will be removed upon next export.

**Review by farmaazon — APPROVED** [review]
LGTM except the changes from the previous PR (see my comment in #9132 )

**Comment by AdRiley** [comment]
Can you share a screenshot of what was previously the design/live mode toggle in the on/off positions?

**Comment by kazcw** [comment]
> Can you share a screenshot of what was previously the design/live mode toggle in the on/off positions?
@AdRiley

Added to description. (The padding is a bit ugly in the Before pics because it's an interim state between moving the project name and this PR).

**Comment by AdRiley** [comment]
Looks good! Thank you :)

**Comment by kazcw** [comment]
I modified the record-once icon to be 16x16 because we're using exclusively 16x16 icons. This might not be exactly the icon design we want, but I'm going to merge it to get the code changes in; we can tweak the icon later.

**Pull request merged**